### DISCUSSION TITLE
Support for Wine in Linux

### DISCUSSION THREAD

**Discussion by fernandomartin777** [musescore/MuseScore#24072] [Ideas]
Many features, sometimes the most outstanding ones, of Musescore 4.x are not available in Linux ad some have no prediction of when they could be implemented. MU4 doesn't work with pipewire sound engine in recent distros. So MU4 in Linux is mostly a notation software instead of a studio. I tried several versions and builds of MU4 for Windows in Wine and it crashed when launching. Could you, please, add a support to run Musescore studio 4 and Musehub under Wine in linux? That way we linux users could have access to the new features in a do-it-yourself way. (Probably if there was not linux version wine community would have added it to winedb, found workarounds to make it work and we would be better than with partially functional native version.)

**Comment by diedeno** [comment]


**Reply by fernandomartin777** [reply]
> Why don't you use Windows?

Windows is very expensive for Brazilian currency and economy standards and it allows very limited customization, almost none.

**Comment by cbjeukendrup** [comment]
Two things:

**I.** The only two features that are missing on Linux are
- Paid Muse Sounds
- VST, but I hope that will come relatively soon because it seems not very difficult to implement (whether any useful VSTs exist on Linux is a different story though, but probably there are some)

For the rest, everything should be available on Linux, or am I missing anything? Perhaps audio driver support needs some fixes, but I doubt that will be any better when using Wine.

**II.** Is there any documentation about what would be necessary to "support" Wine? From their website, I get the impression that one is not supposed to do anything special for supporting Wine; [on the contrary](https://wiki.winehq.org/Developer_FAQ#How_can_I_detect_Wine?). 
Can you get any more information about why/how/where it is crashing? Perhaps we can fix that. 

Stopping to support Linux natively in the hope that the community will get it working in Wine is not an option. The very idea would make too many Linux users too angry.

**Reply by fernandomartin777** [reply]
If you're willing to help making it run under wine, if you tell me how can I find out what crashes I can test and give you feedback. Thanks.

**Reply by MarcSabatella** [reply]
In theory, the main benefit of VST for Linux wouldn’t be third party VST libraries which are indeed uncommon, but rather Muse FX.  Of course, that would also require a commitment from the Muse Sounds team to release that for Linux.

The links you posted earlier about what may be involved in getting VST working in Linux is way outside my comfort zone.  But if we knew for sure that the Muse FX would then be possible, Id be more inclined to give it a shot…

**Reply by cbjeukendrup** [reply]
> If you're willing to help making it run under wine, if you tell me how can I find out what crashes I can test and give you feedback. Thanks.

I'm willing to the extent that if there's something reasonably simple we can do to make it work, I'm happy to do that after working hours. 
But I must admit I barely know anything about Wine, so can't say a lot about how to troubleshoot issues. The first step would be to somehow get a backtrace of the crash; but I wouldn't know if you need to run the Wine MuseScore process inside GDB on Linux, or that you would need to launch WinDbg inside Wine and then start MuseScore from WinDbg. The latter sounds the most viable to me (but again, that's just my gut feeling because I know nothing). So you'd need to get the WinDbg program working in Wine, and inside that program use the "Launch executable" option to start MuseScore4.exe. When it crashes, WinDbg should catch that and show where in MuseScore the crash occurred. If you could upload a screenshot of this, that may be helpful.

**Reply by fernandomartin777** [reply]
> In theory, the main benefit of VST for Linux wouldn’t be third party VST libraries which are indeed uncommon, but rather Muse FX. Of course, that would also require a commitment from the Muse Sounds team to release that for Linux.
> 
> The links you posted earlier about what may be involved in getting VST working in Linux is way outside my comfort zone. But if we knew for sure that the Muse FX would then be possible, Id be more inclined to give it a shot…

I was thinking about some nice vst instruments out there and running them using wine, yabridge etc. For you all who use MU 4 under windows: Can it assign vst instruments to tracks?

**Reply by fernandomartin777** [reply]
> I really don't think the vast majority of Linux users would even think about using Wine if there is a native Linux application available. IMHO, the way forward is to fix any problems with pipewire (since that is the trigger for the OP wanting to use wine). Going down the wine debugging rabbit hole can be a nice hobby of course. (but VERY frustrating and IMHO a complete waste of time. I wasted countless weekends...) The last time i checked, nobody has ever managed running Dorico on Wine, not even with the help of Dorico developers (but i don't really believe they helped a lot...) Nobody managed to run Kontakt 7, Native Access 2 (1.4 works fine), and so on. Frustration all around. Just my idea on the matter of course.

There are good free kontakt instruments out there and I was not able to use them too. Linux is slowly being push out of the music production universe. But imagine a common brazilian worker pay 2/3 of his months wages or more for an original windows license. Not to mention other countries that are in a worse situation. So most people here use piracy without even being aware of the risks.

**Reply by fernandomartin777** [reply]
> > If you're willing to help making it run under wine, if you tell me how can I find out what crashes I can test and give you feedback. Thanks.
> 
> I'm willing to the extent that if there's something reasonably simple we can do to make it work, I'm happy to do that after working hours. But I must admit I barely know anything about Wine, so can't say a lot about how to troubleshoot issues. The first step would be to somehow get a backtrace of the crash; but I wouldn't know if you need to run the Wine MuseScore process inside GDB on Linux, or that you would need to launch WinDbg inside Wine and then start MuseScore from WinDbg. The latter sounds the most viable to me (but again, that's just my gut feeling because I know nothing). So you'd need to get the WinDbg program working in Wine, and inside that program use the "Launch executable" option to start MuseScore4.exe. When it crashes, WinDbg should catch that and show where in MuseScore the crash occurred. If you could upload a screenshot of this, that may be helpful.

Today I found out that 4.4 works with pipewire. There's also a midi out that I didn't have time to test if now I can connect to an sfz player or other kind of midi player. Maybe...
No vst yet.
I don't know if muse sounds manager for linux will keep updating muse sounds correctly. It keeps saying that all my sound are up-to-date. Strangely muse hub official site doesn't allow me not even to download the windows exe to try to see if it works under linux. It says explicitly that my system is not supported. I know I could change my browser agent but I wont waste time with it if devs are willing banning linux users from their new technologies.
Thanks everybody for the help so far.

**Reply by fernandomartin777** [reply]
Thank you. I saw your answer at musescore.org. About kontakt what I had in mind was exactly pianobook. But anyway I give up on kontakt instruments there and use only sfz and decent sampler. If the free features are working on MU 4.4 so probably I won't need to try it on wine anymore.

**Reply by joshuaboniface** [reply]
>I really don't think the vast majority of Linux users would even think about using Wine if there is a native Linux application available.

I'm not all Linux users of course, but from me, that's a definitive "yes".

> The last time i checked, nobody has ever managed running Dorico on Wine, not even with the help of Dorico developers (but i don't really believe they helped a lot...) Nobody managed to run Kontakt 7, Native Access 2 (1.4 works fine), and so on. Frustration all around.

And Sibelius runs like it's on a 100 MHz Pentium 1 CPU.

Wine is for making legacy Windows-only programs run(ish) on Linux. MuseScore is a natively compatible Linux application. If the issue is with Pipewire support, I *very* highly doubt that running the Windows binary under Wine is going to help in *any* way here, and is far more likely to introduce even more bugs and issues (like, say, a crash). And any time contributors spend trying to "tailor" MuseScore for Wine on Linux is time better spent making the native Linux support better.

**Reply by fernandomartin777** [reply]
In 4.4 they added a basic pipewire support that allows it to work again. Sevwral features are being implemented in windows and mac versiona but not in linix version. That's the issue I had in mind at first. Anyway having an incomplete version, a fork, a community version or a trial and error in wine, maybe the result ends the same : we have to live with the idea tgat we won't have a full featured software in linix.

**Reply by MarcSabatella** [reply]
There is to my knowledge one and only one feature implemented for Windows & macOS but not Linux: VST support.  And this will no doubt happen, as the developers have already demonstrated exactly what code needs to be added for this to work (it's not very much).  So it's just waiting for some Linux developer who is interested in seeing this feature to volunteer to do so.  no reason in the world this couldn't happen soon.  Certainly, this would be *way*( simpler than the sorts of handstands that might be required to get all of this working under Wine.

**Comment by tledford4** [comment]
I know that it isn't technically MuseScore Studio at all, but MuseHub, or in the case of Linux, Muse Sounds Manager, that lacks one potentially huge feature that is present on Windows and Mac:  Paid Muse Sounds.  As cbjeukendrup pointed out three weeks ago in an earlier post to this thread.

This is something I would very much like to have on Linux.  But since Muse Sounds Manager is, like MuseHub, proprietary and closed source, I'm assuming that even if I had the required skillset to remedy this, I wouldn't be allowed to.

**Reply by MarcSabatella** [reply]
Since the paid sounds come from third parties, almost non of whom embrace open-source software in any way, it's hard to imagine how open-sourcing Muse Sounds Manager Muse Hub itself would somehow cause these third parties to agree to provide Linux versions of their products.

**Reply by fernandomartin777** [reply]
In steam manywindows games run in linux through proton. Being closed source doesn't prevent sellers to sell to linux users and keep their software closed. But muse hub site download button bans linux users saying the os is not supported. Furthermore muse sounds manager doesn't have any effects nor community muse sounds, neither free nor proprietary.

**Reply by MarcSabatella** [reply]
Indeed, it's not out of the question that a third party might choose to offer their products on Linux.  But obviously, the vast majority of software developers don't do this.  So expecting that the small handful of VST developers who work with Muse Hub to be among that tiny minority of Linux-friendly developers seems pretty unrealistic.  My guess is, even if Muse Hub itself were to *allow* for paid Muse Sounds products on Linux, few if any if the current crop of VST vendors would choose to do so.  They don't make their VST's available on Linux, why would we assume they'd make their Muse Sounds available?

Doesn't mean it wouldn't be nice to see the option made available, but I'd expect very little to come of it.

**Reply by fernandomartin777** [reply]
I don't hope 3rd party devs will write for linux but if vsts worked in linux we could try to use wine + yabridge (or other) to expose windows vsts to MU. That would allow the usage of sfz instruments, pianoteq, truepianos and much more.

**Reply by MarcSabatella** [reply]
Which brings us back to a point made earlier - that Wine really isn't viable for programs of any real complexity.  It either works or it doesn't, and if it doesn't, there isn't likely to be much that can be done about it except to report problems to the Wine developers.

**Comment by tledford4** [comment]
Thank you for making my point more articulately than did I, @fernandomartin777.
